### PR TITLE
add F_pairwise_distance to pnnx  and ncnn

### PR DESCRIPTION
**Pull request by Marsyule**
had added to pnnx

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=4942) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=4942) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=4942) it.</sub>

**Comment by nihui** [comment]
hi, please enable github action in your forked repo, to let code-format job work properly

**Comment by nihui** [comment]
please fix ci error
```
  36/506 Test  #50: test_F_pairwise_distance ......................***Failed    0.06 sec
  File "/data/devops/workspace/tools/pnnx/tests/test_F_pairwise_distance.py", line 52
    return torch.equal(a0,b0) and torch.equal(a1,b1) and torch.equal(a2,b2) and torch.equal(a3,b3)
IndentationError: unexpected indent
CMake Error at /data/devops/workspace/tools/pnnx/tests/run_test.cmake:5 (message):
  Test failed with return value '1'

```

File: tools/pnnx/src/pass_ncnn/F_pairwise_distance.cpp
Comment (nihui) [inline tools/pnnx/src/pass_ncnn/F_pairwise_distance.cpp:1]: There isn't a ncnn layer called PairwiseDistance.
Either implement the layer or drop the ncnn conversion and leave pnnx conversion only.

If you plan to implement the ncnn layer later, add the ncnn conversion at that time.

File: tools/pnnx/src/CMakeLists.txt [hunk 1]
```diff
@@ -154,6 +154,7 @@ set(pnnx_pass_level2_SRCS
     pass_level2/F_mish.cpp
     pass_level2/F_normalize.cpp
     pass_level2/F_pad.cpp
+	pass_level2/F_pairwise_distance.cpp
```
Comment (nihui) [inline tools/pnnx/src/CMakeLists.txt:157]: space instead of tab

File: tools/pnnx/src/CMakeLists.txt [hunk 2]
```diff
@@ -438,6 +439,7 @@ set(pnnx_pass_ncnn_SRCS
     pass_ncnn/F_mish.cpp
     pass_ncnn/F_normalize.cpp
     pass_ncnn/F_pad.cpp
+	pass_ncnn/F_pairwise_distance.cpp
```
Comment (nihui) [inline tools/pnnx/src/CMakeLists.txt:442]: space instead of tab

File: tools/pnnx/src/pass_level2/F_pairwise_distance.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const char* match_pattern_graph() const
+    {
+        return R"PNNXIR(7767517
+7 6
+pnnx.Input              input_0     0 1 x1
+pnnx.Input              input_1     0 1 x2
+prim::Constant          op_0     	0 1 p value=%p
+prim::Constant          op_1    	0 1 eps value=%eps
+prim::Constant          op_2    	0 1 keepdim value=%keepdim
+aten::pairwise_distance op_3     	5 1 x1 x2 p eps keepdim out
```
Comment (nihui) [inline tools/pnnx/src/pass_level2/F_pairwise_distance.cpp:31]: align spaces

**Review by nihui — CHANGES_REQUESTED** [review]
overall it looks good besides some space style issues

File: tools/pnnx/src/pass_ncnn/F_pairwise_distance.cpp
Comment (nihui) [inline tools/pnnx/src/pass_ncnn/F_pairwise_distance.cpp:1]: remove this source as no such ncnn layer exists

**Review by nihui — CHANGES_REQUESTED** [review]

**Review by nihui — APPROVED** [review]

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**